### ISSUE TITLE
Service "http_client" is not registered in the container

### ISSUE THREAD

**Issue by COil** [phpstan/phpstan-symfony#136]
Hi, I have this warning which seem related to the Symfony plugin : 

```
------ -----------------------------------------------------------
  Line   tests/WebTestCase.php
 ------ -----------------------------------------------------------
  94     Service "http_client" is not registered in the container.
 ------ -----------------------------------------------------------
```

I tried with "HttpClientInterface" same result. 

Here is the code : 

```php
        $httpClient = self::$container->get('http_client');
        if (!$httpClient instanceof HttpClientInterface) {
            throw new \RuntimeException('Invalid http_client service.');
        }
```

When debugging the service I have this : 

    ! [NOTE] The "http_client" service or alias has been removed or inlined when the container was compiled.

I have a similar call : 

```php
    return self::$container->get('doctrine.dbal.default_connection');
```

No warning is returned in this case, but I don't have a "alias has been removed" warning neither.

I use Level max and strict rules. It's the last "false positive" I seem to have.

See you.

**Comment by ondrejmirtes** [comment]
Hi, please create a small reproducing repo that shows your problem.

**Comment by COil** [comment]
Here it is:

https://github.com/COil/phpstan-symfony-issues-136/commits/master 

I thought at first that I would not have the same bug with an almost empty project, but no I have the same error.

**Comment by mhujer** [comment]
@COil I guess that it may be related to Symfony using a different container XML in the test environment (See my blogpost https://blog.martinhujer.cz/how-to-configure-phpstan-for-symfony-applications/ - and search the page for _"srcApp_KernelTestDebugContainer"_)

The second idea is that the `http_client` is not injected into any real controller (or anywhere) so it is inlined away for good (and even using the "test" container won't help)

**Comment by COil** [comment]
Yes indeed, thanks for the hint.

**Comment by COil** [comment]
symfony:
        container_xml_path: ../var/cache/test/App_KernelTestDebugContainer.xml
        #container_xml_path: ../var/cache/prod/App_KernelProdDebugContainer.xml

It's indeed OK with this setup or using the prod env.

**Comment by COil** [comment]
```
------ --------------------------------------------------------------------------------------------------------------------------------------------------------
  Line   tests/WebTestCase.php
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------
  95     Instanceof between Symfony\Contracts\HttpClient\HttpClientInterface and Symfony\Contracts\HttpClient\HttpClientInterface will always evaluate to true.
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------

 -- -----------------------------------------------------------------------------------------------------------------------
     Error
 -- -----------------------------------------------------------------------------------------------------------------------
     Ignored error pattern #Service "http_client" is not registered in the container.# was not matched in reported errors.
 -- -----------------------------------------------------------------------------------------------------------------------
```
